### PR TITLE
Silence clippy

### PR DESCRIPTION
**Pull request by notgull**
It appears that clippy's dead code detection has gotten better. This
commit fixes winit's code to match.

- [x] Tested on all platforms changed
- [x] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Review by kchibisov — CHANGES_REQUESTED** [review]
I'd say that we shouldn't add that many `#[allow(dead_code)]` statements. Probably better to use other cfgs if we already adding one.

**Comment by notgull** [comment]
@madsmtm The iOS failure in CI looks spurious to me, am I correct?

**Comment by notgull** [comment]
> I'd say that we shouldn't add that many `#[allow(dead_code)]` statements. Probably better to use other cfgs if we already adding one.

In many cases we can't use `cfg(...)` guards to get around this issue, like for error types.

**Comment by madsmtm** [comment]
> The iOS failure in CI looks spurious to me, am I correct?

I'd guess so

File: src/platform_impl/web/event_loop/state.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@ use web_time::Instant;
 pub enum State {
     Init,
     WaitUntil {
+        #[allow(dead_code)]
         timeout: backend::Schedule,
```
Comment (daxpedda) [inline src/platform_impl/web/event_loop/state.rs:10]:
```suggestion
        _timeout: backend::Schedule,
```
This is stored here for its drop behavior.

File: src/platform_impl/web/event_loop/state.rs [hunk 2]
```diff
@@ -14,6 +15,7 @@ pub enum State {
         start: Instant,
     },
     Poll {
+        #[allow(dead_code)]
         request: backend::Schedule,
```
Comment (daxpedda) [inline src/platform_impl/web/event_loop/state.rs:19]:
```suggestion
        _request: backend::Schedule,
```

**Review by daxpedda — CHANGES_REQUESTED** [review]

**Review by kchibisov — APPROVED** [review]

**Review by daxpedda — APPROVED** [review]

**Pull request merged**